### PR TITLE
(maint) Only require one of config and config defaults

### PR DESCRIPTION
**Pull request by smcclellan**
In order to accommodate both the PE and FOSS versions of razor-server, the
logic for finding the config file should require that only one of the config
and config defaults files exists.

### CODE REVIEW

File: lib/razor/config.rb [hunk 1]
```diff
@@ -24,42 +24,53 @@ class Config
...
       defaults_file ||= ENV["RAZOR_CONFIG_DEFAULTS"] ||
           (File.file?('/opt/puppetlabs/server/apps/razor-server/config-defaults.yaml') and '/opt/puppetlabs/server/apps/razor-server/config-defaults.yaml') ||
           File::join(File::dirname(__FILE__), '..', '..', 'config-defaults.yaml')
-      begin
+
+      # Find the user-maintained config.yaml
+      # Priority: 1) environment variable
+      # 2) default path in /etc
+      # 3) root directory
+      fname ||= ENV["RAZOR_CONFIG"] ||
```
Comment (steveax) [inline lib/razor/config.rb]: I think it'd be clearer if this had a less generic name, maybe: `config_defaults_file` and config_user_file`?

File: lib/razor/config.rb [hunk 2]
```diff
@@ -24,42 +24,53 @@ class Config
...
         yaml = File::open(defaults_file, "r") { |fp| YAML::load(fp) } || {}
 
         @values.merge!(yaml["all"] || {})
         @values.merge!(yaml[Razor.env] || {})
-      rescue Errno::ENOENT
-        # The defaults file does not exist. This is okay.
-      rescue Errno::EACCES
+      elsif File.exists?(defaults_file)
         raise InvalidConfigurationError,
               _("The configuration defaults file %{filename} is not readable") % {filename: defaults_file}
```
Comment (steveax) [inline lib/razor/config.rb]: Is this testing if it's present, or readable?

File: lib/razor/config.rb [hunk 3]
```diff
@@ -24,42 +24,53 @@ class Config
...
-      rescue Errno::ENOENT
-        raise InvalidConfigurationError,
-          _("The configuration file %{filename} does not exist") % {filename: fname}
-      rescue Errno::EACCES
+        @values.merge!(yaml["all"] || {})
+        @values.merge!(yaml[Razor.env] || {})
+      elsif File.exists?(fname)
         raise InvalidConfigurationError,
-          _("The configuration file %{filename} is not readable") % {filename: fname}
+              _("The configuration file %{filename} is not readable") % {filename: fname}
```
Comment (steveax) [inline lib/razor/config.rb]: "is present but not readable"?

File: lib/razor/config.rb (hunk 2 above)
Comment (steveax) [reply lib/razor/config.rb]: ahh, I see, maybe then: "is present but not readable"?

**Comment by steveax** [comment]
Aside from a few nits, 👍

**Comment by smcclellan** [comment]
Already fixed via 1355bc13881b45f93e53584cfc38a949e2349966.

**Pull request closed without merging**